### PR TITLE
Updated documentation for new `alr publish`

### PR DESCRIPTION
**Pull request by mosteo**

### CODE REVIEW

File: doc/publishing.md [hunk 1]
```diff
@@ -49,50 +51,135 @@ Each crate is "owned" by a list of maintainers, provided with the
...
-       through the GitHub web interface (or the
-       [`hub`](https://github.com/github/hub) tool).
-        1. The base branch you select for the pull request will determine where
-           your changes will become available; see the section on [index
-           branches](#index-branches) for details.
+Depending on how you develop your project, you can use one of the following
+methods to prepare your release submission:
+
+### Starting from a git repository that contains an Alire workspace
+
```
Comment (Fabien-Chouteau) [inline doc/publishing.md:68]: For both git options there should be an explanation about the list of allowed host and the fact the the commit must be on the host repo for the process to succeed.

File: doc/publishing.md [hunk 2]
```diff
@@ -49,50 +51,135 @@ Each crate is "owned" by a list of maintainers, provided with the
...
+`alr publish <URL>`
+
+and the publishing process will carry on as in the previous cases, performing
+the checks and providing you with a file to submit to the index, and an upload
+URL to do so.
+
+### Starting with a local source folder
+
+Assisting with the creation an upload of a source archive from local sources, as
```
Comment (Fabien-Chouteau) [inline doc/publishing.md]: `and`?

File: doc/publishing.md [hunk 3]
```diff
@@ -49,50 +51,135 @@ Each crate is "owned" by a list of maintainers, provided with the
...
+
+### Support for complex projects whose sources become multiple Alire crates
+
+In case your project does not easily map to a single Alire crate (e.g., because
+you manage multiple project files with different dependencies, or there are
+other reasons to keep the sources together even if they generate several
+crates), at this time you will need to prepare individual online source
+archives (or repositories) and proceed from there.
+
+This is also an important scenario in our roadmap, so stay tuned for more
```
Comment (Fabien-Chouteau) [inline doc/publishing.md]: I am not convinced that the documentation should make promises roadmap items.

**Review by Fabien-Chouteau — CHANGES_REQUESTED** [review]

File: doc/publishing.md (hunk 3 above)
Comment (mosteo) [reply doc/publishing.md]: OK, dropping these.

File: doc/publishing.md (hunk 1 above)
Comment (mosteo) [reply doc/publishing.md:68]: Do you mean, give a rationale for the list? Or just mention it too in the second git case?

Comment (Fabien-Chouteau) [reply doc/publishing.md:68]: Provide both the list and a rationale.

**Review by Fabien-Chouteau — APPROVED** [review]

**Pull request merged**